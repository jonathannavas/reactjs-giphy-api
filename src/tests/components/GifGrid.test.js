import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs');

describe('Test on GifGrid.js ', () => {

    const category = 'Hola mundo';

    test('Match Snapshot GifGrid.js', ()=>{

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('should show items when data is loaded useFetchGifs', () => {

        const gifs = [{
            id: 'abc',
            url: 'https//localhost/imagen.jpg',
            title: 'Cualquier cosa'
        }] 

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category={ category } /> );
        
        expect( wrapper ).toMatchSnapshot();

        expect( wrapper.find('p').exists() ).toBe( false );       
        expect( wrapper.find( 'GifGridItem' ).length).toBe(gifs.length);

    });



});