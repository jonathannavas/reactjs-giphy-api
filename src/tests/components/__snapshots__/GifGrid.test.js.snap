// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifGrid.js  Match Snapshot GifGrid.js 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Hola mundo
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test on GifGrid.js  should show items when data is loaded useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Hola mundo
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Cualquier cosa"
      url="https//localhost/imagen.jpg"
    />
  </div>
</Fragment>
`;
